### PR TITLE
Module code and structure updated, sync with internal repo 1.1.0

### PR DESCRIPTION
**Pull request by powatagdev**
Hello. This PR includes following changes:
- code was reformatted and module structure updated according to validator advice
- POWAAIM-2129: fixed error handling for address related problems
- POWAAIM-2205: fixed billing address being ignored
- POWAAIM-1888: added error message for empty order
- POWAAIM-1218: propagate order create error
- POWAAIM-1218: updated error code to match 2step and 3step

### CODE REVIEW

**Comment by powatagdev** [comment]
@Quetzacoalt91 ,
please can you review this pull request and give me feedback.
Thanks. Jaroslav

File: classes/helper/PowatagTotAdminTabHelper.php [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+class PowatagTotAdminTabHelper
+{
+    /**
+     * Function to delete admin tabs from a menu with the module name.
+     *
+     * @param string $name name of the module to delete
+     */
+    public static function deleteAdminTabs($name)
+    {
+        Db::getInstance()->Execute('DELETE FROM '._DB_PREFIX_.'tab_lang WHERE id_tab IN (SELECT id_tab FROM '._DB_PREFIX_.'tab WHERE module = "'.pSQL($name).'")');
```
Comment (Mekki78) [inline classes/helper/PowatagTotAdminTabHelper.php]: You should use the PrestaShop Object Model (in your case Tab()) to perform action rathan than communicate directly with the database. Use directly the tab delete action (that will carry multishop).

File: classes/helper/PowatagTotAdminTabHelper.php [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+        Db::getInstance()->Execute('DELETE FROM '._DB_PREFIX_.'tab WHERE module = "'.pSQL($name).'"');
```
Comment (Mekki78) [inline classes/helper/PowatagTotAdminTabHelper.php]: use PrestaShop Object Model

Regards,

File: classes/helper/PowatagTotAdminTabHelper.php [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+                    'id_tab' => '',
+                    'id_parent' => $id_parent,
+                    'class_name' => $tab['className'],
+                    'module' => $tab['module'],
+                    'position' => $tab['position'],
+                    'active' => 1,
+                    );
+
+        /* Insert the data to the tab table*/
+        Db::getInstance()->insert('tab', $data);
```
Comment (Mekki78) [inline classes/helper/PowatagTotAdminTabHelper.php]: use the PrestaShop Object Model

Regards,

File: classes/helper/PowatagTotAdminTabHelper.php [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+        //Get last insert id from db which will be the new tab id
+        $id_tab = Db::getInstance()->Insert_ID();
```
Comment (Mekki78) [inline classes/helper/PowatagTotAdminTabHelper.php]: use the PrestaShop Object Model

Regards,

File: classes/helper/PowatagTotAdminTabHelper.php [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+        //Define tab multi language data
+        foreach (Language::getLanguages(false) as $language) {
+            $data_lang = array(
+                            'id_tab' => $id_tab,
+                            'id_lang' => $language['id_lang'],
+                            'name' => $tab['name'],
+                            );
+
+            // Now insert the tab lang data
+            Db::getInstance()->insert('tab_lang', $data_lang);
```
Comment (Mekki78) [inline classes/helper/PowatagTotAdminTabHelper.php]: use the PrestaShop Object Model

Regards,

File: classes/helper/PowatagTotAdminTabHelper.php [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+    /**
+     * Get the Id of a tab by its class name.
+     *
+     * @param string $name classname to get the ID
+     *
+     * @return int id_tab
+     */
+    public static function getAdminTabIDByClassName($name)
+    {
+        return Db::getInstance()->getValue('SELECT id_tab FROM '._DB_PREFIX_.'tab WHERE class_name = "'.pSQL($name).'"');
```
Comment (Mekki78) [inline classes/helper/PowatagTotAdminTabHelper.php]: use the PrestaShop Object Model

Regards,

**Comment by powatagdev** [comment]
Hello @Mekki78 , I changed way how tabs were managed. Now using Tab model instead of database access.

**Comment by mmoustai** [comment]
Done :+1:

**Pull request merged**